src/IECoreHoudini/SOP_SceneCacheSource.cpp

```diff
@@ -361,73 +437,86 @@ void SOP_SceneCacheSource::loadObjects( const IECoreScene::SceneInterface *scene
+					params.animatedPrimVars.push_back( "basis2" );
+					params.animatedPrimVars.push_back( "basis3" );
 
-				params.hasAnimatedTopology = scene->hasAttribute( SceneCache::animatedObjectTopologyAttribute );
-				params.hasAnimatedPrimVars = scene->hasAttribute( SceneCache::animatedObjectPrimVarsAttribute );
-				if ( params.hasAnimatedPrimVars )
+					object = points;
+				}
+				else if( params.geometryType == TransformPointCloud )
 				{
-					const ConstObjectPtr animatedPrimVarObj = scene->readAttribute( SceneCache::animatedObjectPrimVarsAttribute, 0 );
-					const InternedStringVectorData *animatedPrimVarData = IECore::runTimeCast<const InternedStringVectorData>( animatedPrimVarObj.get() );
-					if ( animatedPrimVarData )
+					object = buildTransformPointCloud( scene, currentTransform, time, params, rootSize, currentPath, true );
+				}
+				else
+				{
+					object = scene->readObject( time );
+
+					params.hasAnimatedTopology = scene->hasAttribute( SceneCache::animatedObjectTopologyAttribute );
+					params.hasAnimatedPrimVars = scene->hasAttribute( SceneCache::animatedObjectPrimVarsAttribute );
+					if ( params.hasAnimatedPrimVars )
 					{
-						const std::vector<InternedString> &values = animatedPrimVarData->readable();
-						params.animatedPrimVars.clear();
-						params.animatedPrimVars.resize( values.size() );
-						std::copy( values.begin(), values.end(), params.animatedPrimVars.begin() );
+						const ConstObjectPtr animatedPrimVarObj = scene->readAttribute( SceneCache::animatedObjectPrimVarsAttribute, 0 );
+						const InternedStringVectorData *animatedPrimVarData = IECore::runTimeCast<const InternedStringVectorData>( animatedPrimVarObj.get() );
+						if ( animatedPrimVarData )
+						{
+							const std::vector<InternedString> &values = animatedPrimVarData->readable();
+							params.animatedPrimVars.clear();
+							params.animatedPrimVars.resize( values.size() );
+							std::copy( values.begin(), values.end(), params.animatedPrimVars.begin() );
+						}
 					}
-				}
 
+				}
 			}
-
-			// modify the object if necessary
-			object = modifyObject( object.get(), params );
-
-			// transform the object unless its an identity
-			if ( currentTransform != Imath::M44d() )
+			else if( params.geometryType == TransformPointCloud )
 			{
-				object = transformObject( object.get(), currentTransform, params );
+				object = buildTransformPointCloud( scene, currentTransform, time, params, rootSize, currentPath, false );
 			}
 
-			// convert the object to Houdini
-			if ( !convertObject( object.get(), name, scene, params ) )
+			if ( object )
 			{
-				addWarning( SOP_MESSAGE, ( "Could not convert " + currentPath + " to Houdini" ).c_str() );
+				// modify the object if necessary
+				object = modifyObject( object.get(), params );
+
+				// transform the object unless its an identity
+				if ( currentTransform != Imath::M44d() )
+				{
+					object = transformObject( object.get(), currentTransform, params );
```

Comment:
If I understand correctly, this means that the "P" primvar will be transformed, so the point will have the transform baked in, _as well as_ having additional primvars saying what that transform was. I don't know enough about either  Houdini or the use cases to know if this is desirable or not. If it's desirable, it seems to imply that the additional primvars are just "along for the ride" and won't affect display in Houdini. But if that's the case, then does it mean that baking kindof precludes workflows where someone might want to process the primvars and write a new transform-only cache from them? Or at least makes it impossible to see what you're doing in such a workflow?

Reply:
Yes John, in case you decide to bake the transform into the point position, that's the case.
Note that you can also load the point cloud in object space thus having no transform baked into the point's position.

That's right round tripping is not straight forward at the moment as it requires you keep both point position and prim var in synced.
Though, the main usage is not round tripping for the moment. Fx is more interested to use this as a helper to modify the same scene cache file but loaded with geometry.

For round tripping, we eventually want to use LOPs and USD.